src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Tests.cs

```diff
@@ -253,5 +254,50 @@ public void SkipRecursionIntoBaseDirectorySymlink()
 
             Assert.Null(reader.GetNextEntry());
         }
+
+        [ConditionalTheory(nameof(CanExtractWithTarTool))]
+        [MemberData(nameof(GetPaxAndGnuTestCaseNames))]
+        public void ResultingArchive_CanBeExtractedByOtherTools(string testCaseName)
+        {
+            if (testCaseName == "specialfiles" && !PlatformDetection.IsSuperUser)
+            {
+                throw new SkipTestException("specialfiles is only supported on Unix and it needs sudo permissions for extraction.");
+            }
+
+            using TempDirectory root = new TempDirectory();
+
+            string archivePath = GetTarFilePath(CompressionMethod.Uncompressed, TestTarFormat.pax.ToString(), testCaseName);
+            string tarExtractedPath = Path.Join(root.Path, "tarExtracted");
+            Directory.CreateDirectory(tarExtractedPath);
+
+            // extract with /usr/bin/tar
+            ExtractWithTarTool(archivePath, tarExtractedPath);
+
+            // create archive with TarFile
+            string dotnetTarArchive = Path.Join(root.Path, "dotnetTarArchive.tar");
+            TarFile.CreateFromDirectory(tarExtractedPath, dotnetTarArchive, includeBaseDirectory: false);
+
+            // extract TarFile's archive with /usr/bin/tar
+            string dotnetTarExtractedPath = Path.Join(root.Path, "dotnetTarExtracted");
+            Directory.CreateDirectory(dotnetTarExtractedPath);
+            ExtractWithTarTool(dotnetTarArchive, dotnetTarExtractedPath);
+
+            // compare results
+            Dictionary<string, FileSystemInfo> expectedEntries = GetFileSystemInfosRecursive(tarExtractedPath).ToDictionary(fsi => fsi.FullName.Substring(tarExtractedPath.Length));
+
+            foreach (var actualEntry in GetFileSystemInfosRecursive(dotnetTarExtractedPath))
+            {
+                string keyPath = actualEntry.FullName.Substring(dotnetTarExtractedPath.Length);
+
+                Assert.True(expectedEntries.TryGetValue(keyPath, out FileSystemInfo expectedEntry), "Entry not expected.");
```

Comment:
Tar always stores the entry paths with the Unix separator `/`. The `expectedEntries` items will contain paths with `\` on Windows. Is this not failing?

Reply:
We are not comparing Tar paths vs Windows/Unix paths. Both `expectedEntries` and `actualEntries` are fetched with FileSystemEnumerable.